realkd/rules.py

```diff
@@ -167,6 +168,10 @@ class Rule:
     >>> r = Rule(female, 1.0, 0.0)
     >>> r(titanic.iloc[0]), r(titanic.iloc[1])
     (0.0, 1.0)
+    >>> r(titanic[titanic.index == 0]), r(titanic[titanic.index == 1])
```

Comment:
REGRESSION:
Rules (and conjunctions etc.), when passed a dataframe with one row will return a one element series, not a float.

This makes some sense to me, but wanted to flag it as a regression. Not sure what the standard is here.

Reply:
Can you please check the behaviour of sklearn for this question? On first glance it seems that there should be shape matching in the following sense:

shape 1 x n -> shape 1 x 1

Only the changed behaviour would satisfy that if I understand you correctly. The one-row/one-column input do generally require some work from the caller also in sklearn. If you just input a 1d-array you are asked to convert it to a matrix.